File: DefaultCodeInputOutput/6.php
<?php
function romanToInt($s) {
    //Write your code here
}

function main() {
    global $argv;

    if (count($argv) != 2) {
        echo json_encode(array("error" => "Usage: php script.php '<json_string>'"));
        return;
    }

    $input_json = $argv[1];
    $data = json_decode($input_json, true);

    if (json_last_error() !== JSON_ERROR_NONE) {
        echo json_encode(array("error" => "Invalid JSON input"));
        return;
    }

    if (!isset($data['roman']) || empty($data['roman'])) {
        echo json_encode(array("error" => "No roman numeral found"));
        return;
    }

    $roman_numeral = $data['roman'];
    $integer_value = romanToInt($roman_numeral);
    $result = array("result" => $integer_value);
    echo json_encode($result);
}

main();
?>

File: DefaultCodeInputOutput/6.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int roman_to_int(const char *s) {
    //Write your code here
}

void main(int argc, char *argv[]) {
    if (argc != 2) {
        fprintf(stderr, "Usage: %s '<json_string>'\n", argv[0]);
        return;
    }

    const char *input_json = argv[1];
    char roman_numeral[100] = {0};
    if (sscanf(input_json, "{\"roman\":\"%[^\"]\"}", roman_numeral) != 1) {
        fprintf(stderr, "{\"error\": \"Invalid JSON input\"}\n");
        return;
    }

    if (strlen(roman_numeral) == 0) {
        fprintf(stderr, "{\"error\": \"No roman numeral found\"}\n");
        return;
    }

    int integer_value = roman_to_int(roman_numeral);
    printf("{\"result\": %d}\n", integer_value);
}
